### PR TITLE
autorelay: fix flaky TestReconnectToStaticRelays

### PR DESCRIPTION
**Pull request by MarcoPolo**
Fixes this flaky test.

### CODE REVIEW

**Review by marten-seemann — DISMISSED** [review]

**Comment by marten-seemann** [comment]
The test is still failing:
```
  === RUN   TestReconnectToStaticRelays
      autorelay_test.go:517: 
          	Error Trace:	D:\a\go-libp2p\go-libp2p\p2p\host\autorelay\autorelay_test.go:517
          	Error:      	Not equal: 
          	            	expected: 0
          	            	actual  : 1
          	Test:       	TestReconnectToStaticRelays
  --- FAIL: TestReconnectToStaticRelays (2.89s)

File: p2p/host/autorelay/autorelay_test.go [hunk 1]
```diff
@@ -457,6 +461,40 @@ func TestIncorrectInit(t *testing.T) {
...
+	t.Helper()
+	delta := 0
+	for {
+		select {
+		case evAny := <-addrUpdated.Out():
+			ev := evAny.(event.EvtLocalAddressesUpdated)
+			for _, updatedAddr := range ev.Removed {
+				if updatedAddr.Action == event.Removed {
+					if _, err := updatedAddr.Address.ValueForProtocol(ma.P_CIRCUIT); err == nil {
+						delta -= 1
```
Comment (marten-seemann) [inline p2p/host/autorelay/autorelay_test.go:474]: Nit:
```suggestion
						delta --
```

File: p2p/host/autorelay/autorelay_test.go [hunk 2]
```diff
@@ -457,6 +461,40 @@ func TestIncorrectInit(t *testing.T) {
...
+						if delta == expectedDelta {
+							return
+						}
+					}
+				}
+			}
+			for _, updatedAddr := range ev.Current {
+				if updatedAddr.Action == event.Added {
+					if _, err := updatedAddr.Address.ValueForProtocol(ma.P_CIRCUIT); err == nil {
+						delta += 1
```
Comment (marten-seemann) [inline p2p/host/autorelay/autorelay_test.go:484]: Nit:
```suggestion
						delta++
```

File: p2p/host/autorelay/autorelay_test.go [hunk 3]
```diff
@@ -457,6 +461,40 @@ func TestIncorrectInit(t *testing.T) {
 	_ = newPrivateNode(t)
 }
 
+func expectDeltaInAddrUpdated(t *testing.T, addrUpdated event.Subscription, expectedDelta int) {
+	t.Helper()
+	delta := 0
+	for {
```
Comment (marten-seemann) [inline p2p/host/autorelay/autorelay_test.go:467]: Is it ok that adding and removing would cancel out here?

**Review by marten-seemann — APPROVED** [review]

File: p2p/host/autorelay/autorelay_test.go (hunk 3 above)
Comment (MarcoPolo) [reply p2p/host/autorelay/autorelay_test.go:467]: I think so? I don't think it'll matter really

**Pull request merged**